google/cloud/aiplatform/featurestore/featurestore.py

```diff
@@ -999,17 +984,204 @@ def batch_serve_to_gcs(
+                                "feature_id_2_1": "feature_id_2_1_feature_value",
+                                "bar": "feature_id_2_2_feature_value",
+                                "timestamp": "2020-01-01T10:00:00.123Z"
+                        ],
+                    )
+
+                Timestamp in each read instance must be millisecond-aligned.
+
+                The columns can be in any order.
+
+                Values in the timestamp column must use the RFC 3339 format,
+                e.g. ``2012-07-30T10:43:17.123Z``.
+
+            pass_through_fields (List[str]):
+                Optional. When not empty, the specified fields in the
+                read_instances source will be joined as-is in the output,
+                in addition to those fields from the Featurestore Entity.
+
+                For BigQuery source, the type of the pass-through values
+                will be automatically inferred. For CSV source, the
+                pass-through values will be passed as opaque bytes.
+
+            feature_destination_fields (Dict[str, str]):
+                Optional. A user defined dictionary to map a feature's fully qualified resource name to
+                its destination field name. If the destination field name is not defined,
+                the feature ID will be used as its destination field name.
+
+                Example:
+                    feature_destination_fields = {
+                        'projects/123/locations/us-central1/featurestores/fs_id/entityTypes/et_id1/features/f_id11': 'foo',
+                        'projects/123/locations/us-central1/featurestores/fs_id/entityTypes/et_id2/features/f_id22': 'bar',
+                     }
+
+        Returns:
+            pd.DataFrame: The pandas DataFrame containing feature values from batch serving.
+
+        """
+        try:
+            from google.cloud import bigquery_storage
+        except ImportError:
+            raise ImportError(
+                f"Google-Cloud-Bigquery-Storage is not installed. Please install google-cloud-bigquery-storage to use "
+                f"{self.batch_serve_to_df.__name__}"
+            )
+
+        try:
+            import pyarrow  # noqa: F401 - skip check for 'pyarrow' which is required when using 'google.cloud.bigquery'
+        except ImportError:
+            raise ImportError(
+                f"Pyarrow is not installed. Please install pyarrow to use "
+                f"{self.batch_serve_to_df.__name__}"
+            )
+
+        try:
+            import pandas as pd
+        except ImportError:
+            raise ImportError(
+                f"Pandas is not installed. Please install pandas to use "
+                f"{self.batch_serve_to_df.__name__}"
+            )
+
+        bigquery_client = bigquery.Client(
+            project=self.project, credentials=self.credentials
+        )
+
+        featurestore_name_components = self._parse_resource_name(self.resource_name)
```

Comment:
It seems like this method should include a `self.wait` call because it's possible they could run methods like `create` and `create_entity_type` with `sync=False` and then call this method. That may be the case else where in these classes.

Reply:
Yeah, that looks right. If so, there are more methods need to include this self.wait() before parsing for resource_name_components.

i.e.
```
def get_entity_type(self, entity_type_id: str) -> "featurestore.EntityType":
        featurestore_name_components = self._parse_resource_name(self.resource_name)
```